metrics/my-em-f1-squad-v2.py

```diff
@@ -0,0 +1,136 @@
+            " or multiple expected target."
+        )
+        self.metric_config = self.get_metrics_configuration(self.id)
+        self.endpoints = self.metric_config.get("endpoints", [])
+        self.configurations = self.metric_config.get("configurations", {})
+
+    def get_metadata(self) -> dict | None:
+        """
+        Retrieves and returns the metadata of the ExactStrMatch class.
+
+        Returns:
+            dict | None: A dictionary containing the 'id', 'name', 'description', 'endpoints' and 'configurations'
+            of the ExactStrMatch class, or None if not applicable.
+        """
+        return {
+            "id": self.id,
+            "name": self.name,
+            "description": self.description,
+            "endpoints": self.endpoints,
+            "configurations": self.configurations,
+        }
+
+    @timeit
+    async def get_results(
+        self, prompts: Any, predicted_results: Any, targets: Any, *args, **kwargs
+    ) -> dict:
+        """
+        Calculates the accuracy of the predicted results by comparing them to the target results.
+
+        Args:
+            prompts (Any): The prompts used for prediction.
+            predicted_results (Any): The predicted results.
+            targets (Any): The target results.
+            *args: Additional positional arguments.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            dict: A dictionary containing the accuracy of the predicted results.
+        """
+        em_scores = 0
+        em_results = []
+        f1_scores = 0
+        f1_results = []
+        total = len(predicted_results)
+
+        for idx, (result, target) in enumerate(zip(predicted_results, targets)):
+            # Check if the target is a single or multiple targets
+            if isinstance(target, str):
+                target = ast.literal_eval(target)
+
+            # Process result
+            result = result.response
+            result = result.strip()
+            if result == 'unanswerable':
+                result = ''
+
+            # Process target
+            target = [t for t in target['text'] if self.normalize_answer(t)]
+            if not target:
+                # For unanswerable questions, only correct answer is empty string
+                target = ['']
+            em = max(self.compute_exact(tg, result) for tg in target)
+            em_scores += em
+            em_results.append(em)
+            f1 = max(self.compute_f1(tg, result) for tg in target)
+            f1_scores += f1
+            f1_results.append(f1)
+
+        return {
+            "em": float(em_scores / total) * 100,
+            "f1": float(f1_scores / total) * 100,
+            "em_results": em_results,
+            "f1_results": f1_results,
+            "grading_criteria": {"f1": float(f1_scores / total) * 100},
+        }
+
+    def compute_exact(self, a_gold, a_pred):
```

Comment:
can we add the function docstrings and also include the type hintings for the input args and the returns